src/Transactions/TransactionFields/LinesField.php

```diff
@@ -59,20 +97,30 @@ public function addLine(TransactionLine $line)
     {
         Assert::isInstanceOf($line, $this->getLineClassName());
 
-        $this->lines[] = $line;
-
+        $this->addLinePerType($line);
```

Comment:
I think the code is more readable if you would just include the code here, instead of creating a new method for it.

Reply:
Now that the `array_key_exists()` checks are gone, this is just 1 line and I've moved it to this method again.